src/types/mod.rs

```diff
@@ -358,9 +358,11 @@ impl<T: FromSql> FromSql for Vec<T> {
             _ => panic!("expected array type"),
         };
 
-        if try!(raw.read_i32::<BigEndian>()) != 1 {
-            return Err(Error::Conversion("array contains too many dimensions".into()));
-        }
+        match try!(raw.read_i32::<BigEndian>()) {
+            0 => return Ok(Vec::new()),
```

Comment:
If I let it keep reading, it hits EOF trying to read `_index_offset`. Seemed easier to just return an empty Vec early.